backend/LexBoxApi/Services/HgService.cs

```diff
@@ -280,17 +280,16 @@ public bool RepoIsLocked(ProjectCode projectCode)
 
     public async Task<DateTimeOffset?> GetLastCommitTimeFromHg(ProjectCode projectCode)
     {
-        var json = await GetCommit(projectCode, "tip");
-        //format is this: [1678687688, offset] offset is
-        var dateArray = json?["entries"]?[0]?["date"].Deserialize<decimal[]>();
-        if (dateArray is null || dateArray.Length != 2 || dateArray[0] <= 0)
-            return null;
-        //offsets are weird. The format we get the offset in is opposite of how we typically represent offsets, eg normally the US has negative
-        //offsets because it's behind UTC. But in other cases the US has positive offsets because time needs to be added to reach UTC.
-        //the offset we get here is the latter but dotnet expects the former so we need to invert it.
-        var offset = (double)dateArray[1] * -1;
-        var date = DateTimeOffset.FromUnixTimeSeconds((long)dateArray[0]).ToOffset(TimeSpan.FromSeconds(offset));
-        return date.ToUniversalTime();
+        var dateStr = await GetTipDate(projectCode);
+        // Format is "1472445535 -25200", two ints separated by a single space.
+        // "0 0" is returned for empty repos (no commits), but we prefer to represent that as null
+        if (dateStr == "0 0") return null;
+        var dateArray = dateStr?.Split();
+        if (dateArray is null || dateArray.Length != 2) return null;
+        // Parse to 64-bit ints so we don't have a year 2038 problem
+        if (!long.TryParse(dateArray[0], out var timestamp)) return null;
+        if (!long.TryParse(dateArray[1], out var offset)) return null;
+        return ConvertHgDate(timestamp, offset);
```

Comment:
now that we have this convert method let's shift all the code related do the date parsing into it. So `GetLastCommitTimeFromHg` would just have 2 method calls one to get the dateStr and one to parse it. `ConvertHgDate` could then return null. Then I'd like the test cases expanded to test the code paths in the Convert function.

Reply:
Done in commit ea83fef1.